python/kfserving/kfserving/api/set_credentials.py

```diff
@@ -0,0 +1,188 @@
+    :param str GCP_CREDS_FILE: The path for the gcp credentials file (required).
+    :return: str  The name of created service account.
+    '''
+
+    check_key_in_args(kwargs, constants.GCP_CREDS_ARG_NAME)
+
+    with open(kwargs[constants.GCP_CREDS_ARG_NAME]) as f:
+        gcp_creds_content = f.read()
+
+    # Try to get GCP creds file name from configmap, set default value then if cannot.
+    gcp_creds_file_name = get_creds_name_from_config_map(
+        'gcsCredentialFileName')
+    if not gcp_creds_file_name:
+        gcp_creds_file_name = constants.GCP_CREDS_FILE_DEFAULT_NAME
+
+    string_data = {gcp_creds_file_name: gcp_creds_content}
+
+    gcp_secret_name = create_secret(
+        namespace=namespace, string_data=string_data)
+
+    gcp_sa_name = create_service_account(
+        secret_name=gcp_secret_name,
+        namespace=namespace)
+
+    return gcp_sa_name
+
+
+def create_aws_credentials(namespace, **kwargs):
+    '''
+    Create AWS Credentails (secret and service account).
+    :param str AWS_ACCESS_KEY_ID: AWS access key ID (required).
+    :param str AWS_SECRET_ACCESS_KEY: AWS secret access key (required).
+    :param str S3_ENDPOINT: S3 settings variables S3_ENDPOINT (optional).
+    :param str AWS_REGION: S3 settings variables AWS_REGION (optional).
+    :param str S3_USE_HTTPS: S3 settings variables S3_USE_HTTPS (optional).
+    :param str S3_VERIFY_SSL: S3 settings variables S3_VERIFY_SSL (optional).
+    :return: str  The name of created service account.
+    '''
+
+    check_key_in_args(kwargs, constants.AWS_ACCESS_KEY_ID)
+    check_key_in_args(kwargs, constants.AWS_SECRET_ACCESS_KEY)
+
+    # Try to get AWS creds name from configmap, set default value then if cannot.
+    aws_access_key_id_name = get_creds_name_from_config_map(
+        's3AccessKeyIDName')
+    if not aws_access_key_id_name:
+        aws_access_key_id_name = constants.AWS_ACCESS_KEY_ID_NAME
+
+    aws_secret_access_key_name = get_creds_name_from_config_map(
+        's3SecretAccessKeyName')
+    if not aws_secret_access_key_name:
+        aws_secret_access_key_name = constants.AWS_SECRET_ACCESS_KEY_NAME
+
+    data = {
+        aws_access_key_id_name: kwargs[constants.AWS_ACCESS_KEY_ID],
+        aws_secret_access_key_name: kwargs[constants.AWS_SECRET_ACCESS_KEY],
+    }
+
+    aws_cred_sets = {
+        'S3_ENDPOINT': constants.KFSERVING_GROUP + "/s3-endpoint",
+        'AWS_REGION': constants.KFSERVING_GROUP + "/s3-region",
+        'S3_USE_HTTPS': constants.KFSERVING_GROUP + "/s3-usehttps",
+        'S3_VERIFY_SSL': constants.KFSERVING_GROUP + "/s3-verifyssl",
+    }
+
+    aws_annotations = {}
+    for key, value in aws_cred_sets.items():
+        if key in kwargs.keys():
+            aws_annotations.update({value: kwargs[key]})
+
+    aws_secret_name = create_secret(
+        namespace=namespace, annotations=aws_annotations, data=data)
+
+    aws_sa_name = create_service_account(
```

Comment:
Can we also support the case where we can attach the secret to existing service account  ?

Reply:
@yuzisun Yes, we can support that cases. Do you think if need to support case that user already created secret, and API to create service account?

Reply:
@yuzisun Yes, the api supports the case. For now, user can specify the `service_account`, if so just attach the created secret with the service account, otherwise will create new one.